src/libraries/System.Collections/tests/Generic/Dictionary/Dictionary.Generic.Tests.cs

```diff
@@ -51,6 +51,16 @@ public void Dictionary_Generic_Constructor_IDictionary_IEqualityComparer(int cou
             Assert.Equal(comparer, copied.Comparer);
         }
 
+        [Theory]
+        [MemberData(nameof(ValidCollectionSizes))]
+        public void Dictionary_Generic_Constructor_IDictionary_Optimized(int count)
+        {
+            Dictionary<TKey, TValue> source = new Dictionary<TKey, TValue>(count);
+            AddToCollection(source, count);
+            Dictionary<TKey, TValue> copied = new Dictionary<TKey, TValue>(source);
```

Comment:
There were no unit tests that copied a dictionary and verified that operations such as Add or Remove work on the copied dictionary. Should I make a new unit test to test these more complicated scenarios or test them in this "Constructor" unit test?